docs/rules/spec-strict-refs.md

```diff
@@ -0,0 +1,100 @@
+# spec-strict-refs
+
+Checks that `$ref` is only used in the locations permitted by the OpenAPI specification.
+
+| OAS | Compatibility |
+| --- | ------------- |
+| 2.0 | ✅            |
+| 3.0 | ✅            |
+| 3.1 | ✅            |
+
+## API design principles
+
+This rule is for spec correctness.
+
+Allows to use the `$ref` keyword exclusively for references to elements that may be inside the component section.
+
+It can be useful when other tools are integrated into the API workflow that demands strict adherence to the specifications.
+
+List of elements the `$ref` can be used with:
+
+- Schema
+- Response
+- Parameter
+- Example
+- RequestBody
+- Header
+- SecurityScheme
+- Link
+- Callback
+- PathItem
+
+## Configuration
+
+To configure the rule, add it to the `rules` object in your configuration file.
+Set the desired [severity](/docs/cli/rules.md#severity-settings) for the rule.
+
+```yaml
+rules:
+  spec-strict-refs: error
+```
+
+## Configuration
+
+| Option   | Type   | Description                              |
+| -------- | ------ | ---------------------------------------- |
+| severity | string | Possible values: `off`, `warn`, `error`. |
```

Comment:
Should we explain the difference between warn and error? What happens for an error vs. a warning?